### PR TITLE
api_server: add SGX support to Amber AS

### PR DESCRIPTION
**Pull request by mythi**

### CODE REVIEW

**Comment by mythi** [comment]
the SGX part tested with Amber, @Lu-Biao to help check TDX

**Comment by Lu-Biao** [comment]
It works well with TDX. @mythi

File: src/api/src/attestation/amber/mod.rs [hunk 1]
```diff
@@ -49,18 +50,18 @@ pub struct Amber {
...
-            bail!("Only implement for tdx now");
+        if tee != Tee::Tdx && tee != Tee::Sgx {
+            bail!("Amber: TEE {tee:?} is not supported.");
         }
         // get quote
         let attestation = serde_json::from_str::<Attestation>(attestation)
             .map_err(|e| anyhow!("Deserialize Attestation failed: {:?}", e))?;
-        let tdx_evidence = serde_json::from_str::<TdxEvidence>(&attestation.tee_evidence)
-            .map_err(|e| anyhow!("Deserialize TDX Evidence failed: {:?}", e))?;
+        let evidence = serde_json::from_str::<AmberTeeEvidence>(&attestation.tee_evidence)
```
Comment (Xynnn007) [inline src/api/src/attestation/amber/mod.rs:59]: We'd better raise an `info` or `warn` here to tell the user that `eventlog` is ignored.

**Review by Xynnn007 — APPROVED** [review]
The patch looks good to me, while we need continue the talk about the architecture of KBS.

File: src/api/src/attestation/amber/mod.rs (hunk 1 above)
Comment (mythi) [reply src/api/src/attestation/amber/mod.rs:59]: With amber, `event_log` needs to be added as a separate field in the appraisal request but it's currently not done so there's no change in functionality with this PR.

**Comment by mythi** [comment]
> The patch looks good to me, while we need continue the talk about the architecture of KBS.

can we get it for the release?

**Review by jialez0 — APPROVED** [review]

**Pull request merged**